string_test.py periodically fails

In test harness [pull request](https://github.com/mhmerrill/arkouda/pull/360), the test

```python
def run_test_index(strings, test_strings, cat):
    # int index
    assert(strings[N//3] == test_strings[N//3])
    assert(cat[N//3] == test_strings[N//3])
    print("int index passed")
```

called from

```python
 def test_index(self):
        run_test_index(self.strings, self.test_strings, self.cat)
```

periodically fails with the following message:

```
=================================================== short test summary info ====================================================
FAILED tests/string_test.py::StringTest::test_index - assert '\\"\\\\@PSTZ 4' == '"\\@PSTZ 4'
================================================= 1 failed, 8 passed in 2.69s ==================================================
```

--- a/arkouda/pdarrayclass.py
+++ b/arkouda/pdarrayclass.py
@@ -26,6 +26,18 @@ def parse_single_value(msg : str) -> object:
     -------
     object numpy scalar         
     """
+    def unescape(s):
+        escaping = False
+        res = ''
+        for c in s:
+            if escaping:
+                res += c
+                escaping = False
+            elif c == '\\':
+                escaping = True
+            else:
+                res += c
+        return res
     dtname, value = msg.split(maxsplit=1)
     mydtype = dtype(dtname)
     if mydtype == bool:
@@ -38,7 +50,7 @@ def parse_single_value(msg : str) -> object:
                               format(mydtype.name, value)))
     try:
         if mydtype == str_:
-            return mydtype.type(value.strip('"'))
+            return mydtype.type(unescape(value.strip('"')))
         return mydtype.type(value)
     except:
         raise ValueError(("unsupported value from server {} {}".\

--- a/arkouda/pdarraycreation.py
+++ b/arkouda/pdarraycreation.py
@@ -78,8 +78,9 @@ def array(a : Union[pdarray,np.ndarray]) -> Union[pdarray, Strings]:
         raise RuntimeError("Only rank-1 pdarrays or ndarrays supported")
     # Check if array of strings
     if a.dtype.kind == 'U':
+        encoded = np.array([elem.encode() for elem in a])
         # Length of each string, plus null byte terminator
-        lengths = np.array([len(elem) for elem in a]) + 1
+        lengths = np.array([len(elem) for elem in encoded]) + 1
         # Compute zero-up segment offsets
         offsets = np.cumsum(lengths) - lengths
         # Allocate and fill bytes array with string segments
@@ -89,8 +90,8 @@ def array(a : Union[pdarray,np.ndarray]) -> Union[pdarray, Strings]:
                                 " which exceeds allowed transfer size. Increase " +
                                 "ak.maxTransferBytes to force.").format(nbytes))
         values = np.zeros(nbytes, dtype=np.uint8)
-        for s, o in zip(a, offsets):
-            for i, b in enumerate(s.encode()):
+        for s, o in zip(encoded, offsets):
+            for i, b in enumerate(s):
                 values[o+i] = b
         # Recurse to create pdarrays for offsets and values, then return Strings object
         return Strings(array(offsets), array(values))

--- a/arkouda/strings.py
+++ b/arkouda/strings.py
@@ -363,6 +363,7 @@ class Strings:
         """
         Peel off one or more delimited fields from each string (similar 
         to string.partition), returning two new arrays of strings.
+        *Warning*: This function is experimental and not guaranteed to work.
 
         Parameters
         ----------
@@ -447,6 +448,7 @@ class Strings:
         """
         Peel off one or more delimited fields from the end of each string 
         (similar to string.rpartition), returning two new arrays of strings.
+        *Warning*: This function is experimental and not guaranteed to work.
 
         Parameters
         ----------
@@ -502,6 +504,7 @@ class Strings:
         """
         Join the strings from another array onto one end of the strings 
         of this array, optionally inserting a delimiter.
+        *Warning*: This function is experimental and not guaranteed to work.
 
         Parameters
         ----------
@@ -565,6 +568,7 @@ class Strings:
         """
         Join the strings from another array onto the left of the strings 
         of this array, optionally inserting a delimiter.
+        *Warning*: This function is experimental and not guaranteed to work.
 
         Parameters
         ----------

--- a/arkouda/strings.py
+++ b/arkouda/strings.py
@@ -426,7 +426,7 @@ class Strings:
             raise TypeError("Times must be integer, not {}".format(type(times)))
         if times < 1:
             raise ValueError("Times must be >= 1")
-        msg = "segmentedEfunc {} {} {} {} {} {} {} {} {} {}".format("peel",
+        msg = "segmentedPeel {} {} {} {} {} {} {} {} {} {}".format("peel",
                             self.objtype,
                             self.offsets.name,
                             self.bytes.name,

--- a/pytest.ini
+++ b/pytest.ini
@@ -15,7 +15,7 @@ testpaths =
     tests/pdarray_creation.py
     tests/security_test.py
     tests/setops_test.py
-#   tests/string_test.py # https://github.com/mhmerrill/arkouda/issues/364
+    tests/string_test.py
     tests/where_test.py
     tests/extrema_test.py
 norecursedirs = .git dist build *egg* tests/deprecated/*

--- a/src/GenSymIO.chpl
+++ b/src/GenSymIO.chpl
@@ -29,7 +29,7 @@ module GenSymIO {
      */
     proc arrayMsg(cmd: string, payload: bytes, st: borrowed SymTab): string {
         var repMsg: string;
-        var (dtypeBytes, sizeBytes, data) = payload.splitMsgToTuple(3);
+        var (dtypeBytes, sizeBytes, data) = payload.splitMsgToTuple(b" ", 3);
         var dtype = str2dtype(try! dtypeBytes.decode());
         var size = try! sizeBytes:int;
         var tmpf:file;

--- a/src/SegmentedArray.chpl
+++ b/src/SegmentedArray.chpl
@@ -456,7 +456,10 @@ module SegmentedArray {
       forall i in offsets.aD {
         // First, check whether string contains enough instances of delimiter to peel
         var hasEnough: bool;
-        if i == high {
+        if oa[i] > D.high {
+          // When the last string(s) is/are shorter than the substr
+          hasEnough = false;
+        } else if i == high {
           hasEnough = ((+ reduce truth) - numHits[oa[i]]) >= times;
         } else {
           hasEnough = (numHits[oa[i+1]] - numHits[oa[i]]) >= times;

--- a/src/SegmentedMsg.chpl
+++ b/src/SegmentedMsg.chpl
@@ -69,36 +69,46 @@ module SegmentedMsg {
   proc segmentedEfuncMsg(cmd: string, payload: bytes, st: borrowed SymTab): string throws {
     var pn = Reflection.getRoutineName();
     var repMsg: string;
+    var (subcmd, objtype, segName, valName, valtype, valStr) = payload.decode().splitMsgToTuple(6);
+    var json = jsonToPdArray(valStr, 1);
+    var val = json[json.domain.low];
+    var rname = st.nextName();
+    select (objtype, valtype) {
+    when ("str", "str") {
+      var strings = new owned SegString(segName, valName, st);
+      select subcmd {
+        when "contains" {
+          var truth = st.addEntry(rname, strings.size, bool);
+          truth.a = strings.substringSearch(val, SearchMode.contains);
+          repMsg = "created "+st.attrib(rname);
+        }
+        when "startswith" {
+          var truth = st.addEntry(rname, strings.size, bool);
+          truth.a = strings.substringSearch(val, SearchMode.startsWith);
+          repMsg = "created "+st.attrib(rname);
+        }
+        when "endswith" {
+          var truth = st.addEntry(rname, strings.size, bool);
+          truth.a = strings.substringSearch(val, SearchMode.endsWith);
+          repMsg = "created "+st.attrib(rname);
+        }
+        otherwise {return notImplementedError(pn, "subcmd: %s, (%s, %s)".format(subcmd, objtype, valtype));}
+      }
+    }
+    otherwise {return notImplementedError(pn, "(%s, %s)".format(objtype, valtype));}
+    }
+    return repMsg;
+  }
+
+proc segmentedPeelMsg(cmd: string, payload: bytes, st: borrowed SymTab): string throws {
+    var pn = Reflection.getRoutineName();
+    var repMsg: string;
     var (subcmd, objtype, segName, valName, valtype, valStr,
          idStr, kpStr, lStr, jsonStr) = payload.decode().splitMsgToTuple(10);
     select (objtype, valtype) {
     when ("str", "str") {
       var strings = new owned SegString(segName, valName, st);
       select subcmd {
-        when "contains" {
-          var json = jsonToPdArray(valStr, 1);
-          var val = json[json.domain.low];
-          var rname = st.nextName();
-          var truth = st.addEntry(rname, strings.size, bool);
-          truth.a = strings.substringSearch(val, SearchMode.contains);
-          repMsg = "created "+st.attrib(rname);
-        }
-        when "startswith" {
-          var json = jsonToPdArray(valStr, 1);
-          var val = json[json.domain.low];
-          var rname = st.nextName();
-          var truth = st.addEntry(rname, strings.size, bool);
-          truth.a = strings.substringSearch(val, SearchMode.startsWith);
-          repMsg = "created "+st.attrib(rname);
-        }
-        when "endswith" {
-          var json = jsonToPdArray(valStr, 1);
-          var val = json[json.domain.low];
-          var rname = st.nextName();
-          var truth = st.addEntry(rname, strings.size, bool);
-          truth.a = strings.substringSearch(val, SearchMode.endsWith);
-          repMsg = "created "+st.attrib(rname);
-        }
         when "peel" {
           var times = valStr:int;
           var includeDelimiter = (idStr.toLower() == "true");

--- a/src/ServerConfig.chpl
+++ b/src/ServerConfig.chpl
@@ -205,4 +205,22 @@ module ServerConfig
       return tup;
     }
 
+    proc bytes.splitMsgToTuple(sep: bytes, param numChunks: int) {
+      var tup: numChunks*bytes;
+      var count = tup.indices.low;
+
+      // fill in the initial tuple elements defined by split()
+      for s in this.split(sep, numChunks-1) {
+        tup(count) = s;
+        count += 1;
+      }
+      // if split() had fewer items than the tuple, fill in the rest
+      if (count < numChunks) {
+        for i in count..numChunks-1 {
+          tup(i) = b"";
+        }
+      }
+      return tup;
+    }
+
 }

--- a/src/arkouda_server.chpl
+++ b/src/arkouda_server.chpl
@@ -201,6 +201,7 @@ proc main() {
                 when "segmentLengths"    {repMsg = segmentLengthsMsg(cmd, payload, st);}
                 when "segmentedHash"     {repMsg = segmentedHashMsg(cmd, payload, st);}
                 when "segmentedEfunc"    {repMsg = segmentedEfuncMsg(cmd, payload, st);}
+                when "segmentedPeel"     {repMsg = segmentedPeelMsg(cmd, payload, st);}
                 when "segmentedIndex"    {repMsg = segmentedIndexMsg(cmd, payload, st);}
                 when "segmentedBinopvv"  {repMsg = segBinopvvMsg(cmd, payload, st);}
                 when "segmentedBinopvs"  {repMsg = segBinopvsMsg(cmd, payload, st);}

--- a/test/UnitTestPeelStick.chpl
+++ b/test/UnitTestPeelStick.chpl
@@ -129,7 +129,7 @@ proc testMessageLayer(substr, n, minLen, maxLen) throws {
   d.stop("make_strings");
   var reqMsg = "peel str %s %s str 1 True True True %jt".format(strings.offsetName, strings.valueName, [substr]);
   writeReq(reqMsg);
-  var repMsg = segmentedEfuncMsg(cmd="segmentedEfunc", payload=reqMsg.encode(), st);
+  var repMsg = segmentedPeelMsg(cmd="segmentedPeel", payload=reqMsg.encode(), st);
   writeRep(repMsg);
   var (loAttribs,lvAttribs,roAttribs,rvAttribs) = repMsg.splitMsgToTuple('+', 4);
   var loname = parseName(loAttribs);

--- a/tests/string_test.py
+++ b/tests/string_test.py
@@ -25,7 +25,6 @@ def run_test_argsort(strings, test_strings, cat):
     catperm = ak.argsort(cat)
     catsorted = cat[catperm].to_ndarray()
     assert((catsorted == npsorted).all())
-    print("argsort passed")
 
 def run_test_unique(strings, test_strings, cat):
     # unique
@@ -39,7 +38,7 @@ def run_test_unique(strings, test_strings, cat):
     npset = set(np.unique(test_strings))
     # When converted to a set, should agree with numpy
     assert(akset == npset)
-    print("unique passed")
+    return akset
 
 def run_test_index(strings, test_strings, cat):
     # int index
@@ -64,6 +63,42 @@ def run_comparison_test(strings, test_strings, cat):
     npinds = (test_strings == test_strings[N//4])
     assert(np.allclose(akinds.to_ndarray(), npinds))
 
+def run_test_in1d(strings, cat, base_words):
+    more_choices = ak.randint(0, UNIQUE, 100)
+    akwords = base_words[more_choices]
+    more_words = akwords.to_ndarray()
+    matches = ak.in1d(strings, akwords)
+    catmatches = ak.in1d(cat, akwords)
+    assert((matches == catmatches).all())
+    # Every word in matches should be in the target set
+    for word in strings[matches].to_ndarray():
+        assert(word in more_words)
+    # Exhaustively find all matches to make sure we didn't miss any
+    inds = ak.zeros(strings.size, dtype=ak.bool)
+    for word in more_words:
+        inds |= (strings == word)
+    assert((inds == matches).all())
+
+def run_test_groupby(strings, cat, akset):
+    g = ak.GroupBy(strings)
+    gc = ak.GroupBy(cat)
+    # Unique keys should be same result as ak.unique
+    assert(akset == set(g.unique_keys.to_ndarray()))
+    assert(akset == set(gc.unique_keys.to_ndarray()))
+    assert((gc.permutation == g.permutation).all())
+    permStrings = strings[g.permutation].to_ndarray()
+    # Check each group individually
+    lengths = np.diff(np.hstack((g.segments.to_ndarray(), np.array([g.size]))))
+    for uk, s, l in zip(g.unique_keys.to_ndarray(),
+                        g.segments.to_ndarray(),
+                        lengths):
+        # All values in group should equal key
+        assert((permStrings[s:s+l] == uk).all())
+        # Key should not appear anywhere outside of group
+        assert(not (permStrings[:s] == uk).any())
+        assert(not (permStrings[s+l:] == uk).any())
+
+
 def run_test_contains(strings, test_strings, delim):
     found = strings.contains(delim).to_ndarray()
     npfound = np.array([s.count(delim) > 0 for s in test_strings])
@@ -78,111 +113,8 @@ def run_test_ends_with(strings, test_strings, delim):
     found = strings.endswith(delim).to_ndarray()
     npfound = np.array([s.endswith(delim) for s in test_strings])
     assert((found == npfound).all())
-    
-if __name__ == '__main__':
-    if len(sys.argv) > 1:
-        ak.connect(server=sys.argv[1], port=sys.argv[2])
-    else:
-        ak.connect()
 
-    # with open(__file__, 'r') as f:
-    #     base_words = np.array(f.read().split())
-    # test_strings = np.random.choice(base_words, N, replace=True)
-    # strings = ak.array(test_strings)
-
-    base_words1 = ak.random_strings_uniform(0, 10, UNIQUE, characters='printable')
-    base_words2 = ak.random_strings_lognormal(2, 0.25, UNIQUE, characters='printable')
-    base_words = ak.concatenate((base_words1, base_words2))
-    np_base_words = np.hstack((base_words1.to_ndarray(), base_words2.to_ndarray()))
-    assert(compare_strings(base_words.to_ndarray(), np_base_words))
-    choices = ak.randint(0, base_words.size, N)
-    strings = base_words[choices]
-    test_strings = strings.to_ndarray()
-    cat = ak.Categorical(strings)
-    print("strings =", strings)
-    print("categorical =", cat)
-    print("Generation and concatenate passed")
-  
-    # int index
-    run_test_index(strings, test_strings, cat)
-    print("int index passed")
-  
-    # slice
-    run_test_slice(strings, test_strings, cat)
-    print("slice passed")
-    
-    # pdarray int index
-    run_test_pdarray_index(strings, test_strings, cat)
-    print("pdarray int index passed")
-
-    # comparison
-    run_comparison_test(strings, test_strings, cat)
-    print("comparison passed")
-
-    # pdarray bool index
-    test_pdarray_index(strings, test_strings, cat)
-    print("pdarray bool index passed")
-
-    # in1d and iter
-    # more_words = np.random.choice(base_words, 100)
-    # akwords = ak.array(more_words)
-    more_choices = ak.randint(0, UNIQUE, 100)
-    akwords = base_words[more_choices]
-    more_words = akwords.to_ndarray()
-    matches = ak.in1d(strings, akwords)
-    catmatches = ak.in1d(cat, akwords)
-    assert((matches == catmatches).all())
-    # Every word in matches should be in the target set
-    for word in strings[matches]:
-        assert(word in more_words)
-    # Exhaustively find all matches to make sure we didn't miss any
-    inds = ak.zeros(strings.size, dtype=ak.bool)
-    for word in more_words:
-        inds |= (strings == word)
-    assert((inds == matches).all())
-    print("in1d and iter passed")
-
-    # argsort
-    test_argsort(strings, test_strings, cat)
-    
-    # unique
-    test_unique(strings, test_strings, cat)
-
-    # groupby
-    g = ak.GroupBy(strings)
-    gc = ak.GroupBy(cat)
-    # Unique keys should be same result as ak.unique
-    assert(akset == set(g.unique_keys.to_ndarray()))
-    assert(akset == set(gc.unique_keys.to_ndarray()))
-    assert((gc.permutation == g.permutation).all())
-    permStrings = strings[g.permutation]
-    # Check each group individually
-    lengths = np.diff(np.hstack((g.segments.to_ndarray(), np.array([g.size]))))
-    for uk, s, l in zip(g.unique_keys, g.segments, lengths):
-        # All values in group should equal key
-        assert((permStrings[s:s+l] == uk).all())
-        # Key should not appear anywhere outside of group
-        assert(not (permStrings[:s] == uk).any())
-        assert(not (permStrings[s+l:] == uk).any())
-    print("groupby passed")
-
-    # substring functions
-    x, w = tuple(zip(*Counter(''.join(base_words)).items()))
-    delim = np.random.choice(x, p=(np.array(w)/sum(w)))
-
-    # contains
-    run_test_contains(strings, test_strings, delim)
-    print("contains passed")
-
-    # startswith
-    run_test_starts_with(strings, test_strings, delim)
-    print("startswith passed")
-
-    # endswith
-    run_test_ends_with(strings, test_strings, delim)
-    print("endswith passed")
-
-    # peel
+def run_test_peel(strings, test_strings, delim):
     import itertools as it
     tf = (True, False)
     def munge(triple, inc, part):
@@ -239,11 +171,8 @@ if __name__ == '__main__':
         ltest, rtest = rmunge(triples, inc, part)
         assert((ltest == ls.to_ndarray()).all() and (rtest == rs.to_ndarray()).all())
 
-    print("peel passed")
-
-    # stick
-
-    test_strings2 = np.random.choice(base_words, N, replace=True)
+def run_test_stick(strings, test_strings, base_words, delim):
+    test_strings2 = np.random.choice(base_words.to_ndarray(), N, replace=True)
     strings2 = ak.array(test_strings2)
     stuck = strings.stick(strings2, delimiter=delim).to_ndarray()
     tstuck = np.array([delim.join((a, b)) for a, b in zip(test_strings, test_strings2)])
@@ -254,6 +183,91 @@ if __name__ == '__main__':
     tlstuck = np.array([delim.join((b, a)) for a, b in zip(test_strings, test_strings2)])
     assert ((lstuck == tlstuck).all())
     assert ((strings2 + strings) == strings.lstick(strings2, delimiter="")).all()
+
+        
+if __name__ == '__main__':
+    import sys
+    if len(sys.argv) > 1:
+        ak.connect(server=sys.argv[1], port=sys.argv[2])
+    else:
+        ak.connect()
+
+    # with open(__file__, 'r') as f:
+    #     base_words = np.array(f.read().split())
+    # test_strings = np.random.choice(base_words, N, replace=True)
+    # strings = ak.array(test_strings)
+
+    base_words1 = ak.random_strings_uniform(0, 10, UNIQUE, characters='printable')
+    base_words2 = ak.random_strings_lognormal(2, 0.25, UNIQUE, characters='printable')
+    base_words = ak.concatenate((base_words1, base_words2))
+    np_base_words = np.hstack((base_words1.to_ndarray(), base_words2.to_ndarray()))
+    assert(compare_strings(base_words.to_ndarray(), np_base_words))
+    choices = ak.randint(0, base_words.size, N)
+    strings = base_words[choices]
+    test_strings = strings.to_ndarray()
+    cat = ak.Categorical(strings)
+    print("strings =", strings)
+    print("categorical =", cat)
+    print("Generation and concatenate passed")
+  
+    # int index
+    run_test_index(strings, test_strings, cat)
+    print("int index passed")
+  
+    # slice
+    run_test_slice(strings, test_strings, cat)
+    print("slice passed")
+    
+    # pdarray int index
+    run_test_pdarray_index(strings, test_strings, cat)
+    print("pdarray int index passed")
+
+    # comparison
+    run_comparison_test(strings, test_strings, cat)
+    print("comparison passed")
+
+    # pdarray bool index
+    run_test_pdarray_index(strings, test_strings, cat)
+    print("pdarray bool index passed")
+
+    # in1d and iter
+    # more_words = np.random.choice(base_words, 100)
+    # akwords = ak.array(more_words)
+    run_test_in1d(strings, cat, base_words)
+    print("in1d and iter passed")
+
+    # argsort
+    run_test_argsort(strings, test_strings, cat)
+    
+    # unique
+    akset = run_test_unique(strings, test_strings, cat)
+
+    # groupby
+    run_test_groupby(strings, cat, akset)
+    print("groupby passed")
+
+    # substring functions
+    x, w = tuple(zip(*Counter(''.join(base_words.to_ndarray())).items()))
+    delim = np.random.choice(x, p=(np.array(w)/sum(w)))
+
+    # contains
+    run_test_contains(strings, test_strings, delim)
+    print("contains passed")
+
+    # startswith
+    run_test_starts_with(strings, test_strings, delim)
+    print("startswith passed")
+
+    # endswith
+    run_test_ends_with(strings, test_strings, delim)
+    print("endswith passed")
+
+    # peel
+    run_test_peel(strings, test_strings, delim)
+    print("peel passed")
+
+    # stick
+    run_test_stick(strings, test_strings, base_words, delim)
     print("stick passed")
 
 class StringTest(ArkoudaTest):
@@ -270,15 +284,22 @@ class StringTest(ArkoudaTest):
         self.cat = ak.Categorical(self.strings)
         x, w = tuple(zip(*Counter(''.join(self.base_words.to_ndarray())).items()))
         self.delim =  np.random.choice(x, p=(np.array(w)/sum(w)))
+        self.akset = set(ak.unique(self.strings).to_ndarray())
 
     def test_compare_strings(self):
         assert compare_strings(self.base_words.to_ndarray(), self.np_base_words)
     
     def test_argsort(self):
         run_test_argsort(self.strings, self.test_strings, self.cat)
-        
+
+    def test_in1d(self):
+        run_test_in1d(self.strings, self.cat, self.base_words)
+                      
     def test_unique(self):
         run_test_unique(self.strings, self.test_strings, self.cat)
+
+    def test_groupby(self):
+        run_test_groupby(self.strings, self.cat, self.akset)
     
     def test_index(self):
         run_test_index(self.strings, self.test_strings, self.cat)
@@ -297,3 +318,9 @@ class StringTest(ArkoudaTest):
 
     def test_ends_with(self):
         run_test_ends_with(self.strings, self.test_strings, self.delim)
+
+    def test_peel(self):
+        run_test_peel(self.strings, self.test_strings, self.delim)
+
+    def test_stick(self):
+        run_test_stick(self.strings, self.test_strings, self.base_words, self.delim)

--- a/tests/string_test.py
+++ b/tests/string_test.py
@@ -20,7 +20,7 @@ errors = False
 def run_test_argsort(strings, test_strings, cat):
     akperm = ak.argsort(strings)
     aksorted = strings[akperm].to_ndarray()
-    npsorted = np.sort(strings)
+    npsorted = np.sort(test_strings)
     assert((aksorted == npsorted).all())
     catperm = ak.argsort(cat)
     catsorted = cat[catperm].to_ndarray()
@@ -268,7 +268,7 @@ class StringTest(ArkoudaTest):
         self.strings = self.base_words[choices]
         self.test_strings = self.strings.to_ndarray()
         self.cat = ak.Categorical(self.strings)
-        x, w = tuple(zip(*Counter(''.join(self.base_words)).items()))
+        x, w = tuple(zip(*Counter(''.join(self.base_words.to_ndarray())).items()))
         self.delim =  np.random.choice(x, p=(np.array(w)/sum(w)))
 
     def test_compare_strings(self):

--- a/tests/string_test.py
+++ b/tests/string_test.py
@@ -2,6 +2,7 @@ import numpy as np
 from collections import Counter
 from context import arkouda as ak
 from base_test import ArkoudaTest
+import pytest
 ak.verbose = False
 
 N = 100
@@ -197,13 +198,14 @@ if __name__ == '__main__':
     # test_strings = np.random.choice(base_words, N, replace=True)
     # strings = ak.array(test_strings)
 
-    base_words1 = ak.random_strings_uniform(0, 10, UNIQUE, characters='printable')
+    base_words1 = ak.random_strings_uniform(1, 10, UNIQUE, characters='printable')
     base_words2 = ak.random_strings_lognormal(2, 0.25, UNIQUE, characters='printable')
-    base_words = ak.concatenate((base_words1, base_words2))
+    gremlins = ak.array([' ', ''])
+    base_words = ak.concatenate((base_words1, base_words2, gremlins))
     np_base_words = np.hstack((base_words1.to_ndarray(), base_words2.to_ndarray()))
     assert(compare_strings(base_words.to_ndarray(), np_base_words))
     choices = ak.randint(0, base_words.size, N)
-    strings = base_words[choices]
+    strings = ak.concatenate((base_words[choices], gremlins))
     test_strings = strings.to_ndarray()
     cat = ak.Categorical(strings)
     print("strings =", strings)
@@ -274,12 +276,13 @@ class StringTest(ArkoudaTest):
   
     def setUp(self):
         ArkoudaTest.setUp(self)
-        base_words1 = ak.random_strings_uniform(0, 10, UNIQUE, characters='printable')
+        base_words1 = ak.random_strings_uniform(1, 10, UNIQUE, characters='printable')
         base_words2 = ak.random_strings_lognormal(2, 0.25, UNIQUE, characters='printable')
-        self.base_words = ak.concatenate((base_words1, base_words2))
+        gremlins = ak.array([' ', ''])
+        self.base_words = ak.concatenate((base_words1, base_words2, gremlins))
         self.np_base_words = np.hstack((base_words1.to_ndarray(), base_words2.to_ndarray()))
         choices = ak.randint(0, self.base_words.size, N)
-        self.strings = self.base_words[choices]
+        self.strings = ak.concatenate((self.base_words[choices], gremlins))
         self.test_strings = self.strings.to_ndarray()
         self.cat = ak.Categorical(self.strings)
         x, w = tuple(zip(*Counter(''.join(self.base_words.to_ndarray())).items()))
@@ -319,8 +322,10 @@ class StringTest(ArkoudaTest):
     def test_ends_with(self):
         run_test_ends_with(self.strings, self.test_strings, self.delim)
 
+    @pytest.mark.skip(reason="awaiting bug fix.")
     def test_peel(self):
         run_test_peel(self.strings, self.test_strings, self.delim)
 
+    @pytest.mark.skip(reson="awaiting bug fix.")
     def test_stick(self):
         run_test_stick(self.strings, self.test_strings, self.base_words, self.delim)

